pyiceberg/table/__init__.py

```diff
@@ -1750,7 +1750,7 @@ def to_arrow_batch_reader(self) -> pa.RecordBatchReader:
         return pa.RecordBatchReader.from_batches(
             target_schema,
             batches,
-        )
+        ).cast(target_schema)
```

Comment:
This will still return `large` types if you stream the batches because we don't want to fetch all the schemas upfront.